### ISSUE TITLE
Getting Cannot read property 'description' of undefined

### ISSUE THREAD

**Issue by NivedhaSenthil** [getgauge/taiko#353]
### Expected behavior
`evaluate` should give appropriate results always

### Actual behavior
> openBrowser()
 ✔ Browser opened
> evaluate(()=>{return window.location.toString()})
" ✘ Error: Cannot read property 'description' of undefined, run `.trace` for more info."
> goto("google.com")
 ✔ Navigated to url "http://google.com"
> evaluate(()=>{return window.location.toString()})
 ✔ Evaluated given script. Result: https://www.google.com/?gws_rd=ssl

`Error: Cannot read property 'description' of undefined` is thrown when trying to get url of new tab, though executing the same statement on console gives `"about:blank"`. URL of others are given properly.

**Comment by hrmeetsingh** [comment]
Hi team,
I was trying a workaround for adding a listener on page errors (issue [https://github.com/getgauge/taiko/issues/322](url) ).

`await evaluate(() => {
    window.onerror = (message, source, lineno, colno, error) => {
      console.log(message)
    };
  });`

getting the same issue - `Error: Cannot read property 'description' of undefined`

Any update on this?

**Comment by saikrishna321** [comment]
@hrmeetsingh @NivedhaSenthil 

https://github.com/saikrishna321/taiko-diagnostics#logconsoleinfo-command 

I have implemented this feature in taiko-diagnostics. Works fine in headful mode, not in headless mode. 

@NivedhaSenthil  Do you know why it does not work on headless mode?

**Comment by NivedhaSenthil** [comment]
I could successfully get error messages in both headless and headful mode, with the below sample. 
Hope this is helpful.
```
(async () => {
    try {
        await openBrowser();
        await client().Log.enable();
        await client().Log.entryAdded(console.log);
        await goto('gauge.org/sdfsd');
        await reload();
    } catch (e) {
        console.error(e);
    } finally {
        await closeBrowser();
    }
})();
```

**Comment by hrmeetsingh** [comment]
@NivedhaSenthil  - thanks, this worked for me in both modes and am using this for now. Is there any future plans of exposing more devtools domains/methods?

**Comment by NivedhaSenthil** [comment]
Devtools domains/methods are exposed whenever they are need by Taiko api. Please file issues with relevant use cases for any domains to be exposed and comment on existing issues so that it gets prioritised. For logs please track #322

**Comment by saikrishna321** [comment]
@NivedhaSenthil 

can you try the below example with headless and headful

```
    await client().Log.enable();
    await client().Log.entryAdded(console.log);
    await goto('http://note.ly');
    await click('Login');
```

**Comment by NivedhaSenthil** [comment]
ya weird could see log not coming up in headless mode for this. Can you please create a new issue to discuss this as this is off topic from the actual issue

**Comment by luciferankon** [comment]
This is working on commit d591dcfae8416d6e9b35fc9dc2f9d6d9ac7d86de

**Comment by Debashis9012** [comment]
This issue has been verified and found fixed with the commit: b3207ffb094e44f98b22febddf02e84c258a638f

**Comment by wanderingeye8** [comment]
@nehashri My test case failed and shows the error "TypeError: Cannot read property 'exceptionDetails' of undefined". Do you have any idea why this message is shown instead of the actual cause?
